array-manipulation-3/src/difference.js

```diff
@@ -0,0 +1,32 @@
+/* exported difference */
+function difference(first, second) {
+  const symmetricDiff = {};
+  const res = [];
+  for (const word of first) {
+    if (word in symmetricDiff) {
+      delete symmetricDiff[word];
+    } else {
+      symmetricDiff[word] = true;
+    }
+  }
```

Comment:
If a word appears multiple times in the one `Array` and not-at-all in the other `Array`, the `return`ed `Array` should still contain a copy of the word.

---

![image](https://user-images.githubusercontent.com/16143771/150583754-bb33353a-fe89-4d5b-aacd-9356e09e2002.png)